src/reports/components/report-sections/report-section-factory.tsx

```diff
@@ -29,6 +30,7 @@ export type SectionProps = {
     userConfigurationStoreData: UserConfigurationStoreData;
     targetAppInfo: TargetAppData;
     shouldAlertFailuresCount?: boolean;
+    footerText: ReportFooterText;
```

Comment:
This feels odd... why not go through `ReportSectionFactory.FooterSection` directly? you can wrap both the "default" behavior (as in, the current AI-Web behavior) and the package behavior in a common component and change the value of `FooterSection` when the instance of `ReportSectionFactory` is created, which is a different place in each case.

Reply:
Yeah. Maybe.

Reply:
I was trying to avoid shimming it like that.

Reply:
Ok, I have an idea that might be more open-closed. This is taking forever, but I'm having fun.

Reply:
Ok, I did the shim or decorator approach and I think it's an improvement. @smoralesd -- thoughts?